flaml/autogen/math/math_solver.py

```diff
@@ -0,0 +1,513 @@
+Please format the query in json:
+{ "tool" : "python",
+"query": "" # your code here.
+}
+4. Wait for me to give the results.
+5. Correct this step based on the results, or give a new query if the results are invalid.
+6. When you get the answer, put the answer in \\boxed{}.
+""",
+    # use wolfram
+    "wolfram": """Let's use Wolfram Alpha to solve a math problem step by step. You should always follow your own reasoning and only query when necessary.
+
+First state the key idea to solve the problem. Then follow the process:
+1. Output one step.
+2. Take out any queries that can be asked through Wolfram Alpha (for example, any calculations or equations that can be calculated).
+Please format the query in json:
+{ "tool" : "wolfram",
+"query": "" # your query here. Please use wolfram language.
+}
+4. Wait for me to give the results.
+5. Correct this step based on the results, or give a new query if the results are invalid.
+6. When you get the answer, put the answer in \\boxed{}.
+""",
+}
+
+
+class MathSolver:
+    def __init__(
+        self,
+        model,
+        prompt_type="select",
+        prompt_location="user",
+        max_round=10,
+        max_invalid_q_per_step=3,
+        n=1,
+        temperature=1,
+        logger=None,
+        use_cache=True,
+    ):
+        self.max_round = max_round
+        if prompt_type not in PROMPTS:
+            raise ValueError(f"Tool {prompt_type} not supported, choose from {PROMPTS.keys()}")
+
+        self.prompt_type = prompt_type
+        self.prompt_loaction = prompt_location
+        self.prompt = PROMPTS[prompt_type]
+        messages = (
+            [{"role": "system", "content": self.prompt}]
+            if prompt_location == "system"
+            else [{"role": "system", "content": "You are a helpful assistant."}]
+        )
+        self.deafult_config = {
+            "model": model,
+            "messages": messages,
+            "n": n,  # n should be 1 for now
+            'temperature' : temperature,
+        }
+
+        self.max_invalid_q_per_step = max_invalid_q_per_step
+        self.use_cache = use_cache
+        self.logger = logger
+
+    def make_conversation(self, problem, n=1, file_to_be_saved=None):
+        # initialize the query handler
+        query_handler = QueryHandler()
+
+        # initialize the conversation
+        config = copy.deepcopy(self.deafult_config)
+        problem_prompt = {
+            "role": "user",
+            "content": self.prompt + "\nProblem: " + remove_asy_sections(problem["problem"]),
+        }  # put prompt in user message
+        if self.prompt_loaction == "system":
+            problem_prompt = {"role": "user", "content": remove_asy_sections(problem["problem"])}
```

Comment:
Is this correct? The if condition says "system"

Reply:
Thanks for your revision.  This line is saying the if self.prompt_loaction is 'system', the prompt is already in the system message, so here the user prompt is changed to only the question instead of prompt+question.